src/controllers/references.ts

```diff
@@ -30,7 +31,33 @@ export const getReferencesByType = async (
 ) => {
   try {
     const { type } = req.params;
+    const { populate } = req.query;
     const data = await DefinedReference.find({ type });
+
+    if (type === "roles" && populate) {
+      await Promise.all(
+        data.map(async (role) => {
+          const roleJsonldData = JSON.parse(role.jsonld);
+          const test = await DefinedReference.find(
```

Comment:
Change test variable name.
Might also be interesting to .lean() the DefinedReference.find() to get the object directly

Reply:
added lean() and renamed variable